crucible-llvm/test/MemSetup.hs

```diff
@@ -74,11 +79,12 @@ withLLVMCtx mod' action =
           let ?ptrWidth = width
           let ?lc = LLVMTr._llvmTypeCtx ctx
           let ?recordLLVMAnnotation = \_ _ _ -> pure ()
-          action ctx sym
+          action ctx bak
         }}}
   in withIONonceGenerator $ \nonceGen ->
      withHandleAllocator  $ \halloc   -> with nonceGen halloc
 
+data CrucibleLLVMTestData t = CrucibleLLVMTestData
```

Comment:
Previously the "backend" data structures occupied the "user data" slot of a What4 expression builder.  We have now lifted those data structures out separately, so now that user data slot is unoccupied; but something has to go there when calling `newExprBuilder`.  Maybe it's better to add a `EmptyUserData` type to `What4.Expr.Builder` instead of separate ones at use sites.  There ended up being more of them than I expected.